buildbot-conf/cordova.conf

```diff
@@ -0,0 +1,336 @@
+
+# constants
+DEFAULT_REPO_NAME = 'src'
+BASE_WORKDIR      = '.'
+TEST_APP_NAME     = 'mobilespec'
+
+OSX     = 'osx'
+LINUX   = 'linux'
+WINDOWS = 'windows'
+
+# patterns
+CORDOVA_REPO_PATTERN = r'^.*(cordova-[^\.]+)\.git$'
+
+# interpretation of every byte-sized return code as success
+ALWAYS_SUCCESS = {i: SUCCESS for i in range(0, 256)}
+
+####### UTILITIES
+
+# helper functions
+def cordova_builders():
+    return [b.name for b in c['builders'] if b.name.startswith('cordova-')]
+
+def repo_name_from_url(url):
+    match = re.match(CORDOVA_REPO_PATTERN, url)
+    if match is not None:
+        return match.group(1)
+    return DEFAULT_REPO_NAME
+
+def repo_codebase_from_name(name):
+    repo          = projects_config[name]
+    codebase_name = repo['codebase']
+    return repo['codebases'][codebase_name]
+
+def repo_url_from_name(name):
+    return repo_codebase_from_name(name)['repo']
+
+def repo_branch_from_name(name):
+    return repo_codebase_from_name(name)['branch']
+
+def slugify(string):
+    return string.replace(' ', '-')
+
+def running_tasks_on_platform(platform_name, os_name):
+    """
+    Return the names of tasks possibly started by
+    builds on the given platform and OS.
+    """
+    if platform_name == 'windows':
+        return ['WWAHost.exe']
+    elif platform_name == 'wp8':
+        return ['Xde.exe']
+    elif platform_name == 'ios':
+        return ['iOS Simulator']
+    return []
+
+def can_find_running_tasks(step):
+    """
+    Return true if an OS and a platform is specified. Those are the
+    criteria for finding a task because running_tasks_on_platform uses
+    those properties to determine which tasks could be running.
+    """
+    return (
+        (step.build.getProperty('slaveos') is not None) and
+        (step.build.getProperty('platform') is not None)
+    )
+
+# renderers
+@renderer
+def render_repo_name(props):
+    repo_url = props.getProperty('repository')
+    return repo_name_from_url(repo_url)
+
+@renderer
+def render_task_kill_command(props):
+
+    os_name       = props.getProperty('slaveos')
+    platform_name = props.getProperty('platform')
+    running_tasks = running_tasks_on_platform(platform_name, os_name)
+
+    if not running_tasks:
+        return ['echo', 'No tasks to kill known.']
+
+    if os_name == WINDOWS:
+        command = ['taskkill', '/F']
+        for task in running_tasks:
+            command.append('/IM')
+            command.append(task)
+
+    else:
+        command = ['killall']
+        command.extend(running_tasks)
+
+    return command
+
+@renderer
+def render_run_args(props):
+    platform = props.getProperty('platform')
+    if platform == 'windows':
+        return '--win'
+    return ''
+
+# step wrappers
+def DescribedStep(step_class, description, haltOnFailure=True, **kwargs):
+    return step_class(description=description, descriptionDone=description, name=slugify(description), haltOnFailure=haltOnFailure, **kwargs)
```

Comment:
Sometimes name and description are used together i.e. name and description are concatenated (e.g. step description at build details page). 

Your code sets nearly the same value to step description and name. That's why user sees step names with doubled description like `npm-installing-cordova-medic npm installing cordova-medic` (please note `npm installing cordova-medic` is written twice).

In cases when description is long the build step description shown to user is two times longer.

I propose to shorten name somehow: get first 10 characters of description or add special parameter for name.